src/careamics_restoration/engine.py

```diff
@@ -91,14 +91,18 @@ def __init__(
             assert config_path is not None, "config_path is None"  # mypy
             self.cfg = load_configuration(config_path)
 
-        # Create model, optimizer, lr scheduler and gradient scaler
+        # get device, CPU or GPU
+        self.device = get_device()
+
+        # Create model, optimizer, lr scheduler and gradient scaler and load everything
+        # to the specified device
         (
             self.model,
             self.optimizer,
             self.lr_scheduler,
             self.scaler,
             self.cfg,
-        ) = create_model(config=self.cfg, model_path=model_path)
+        ) = create_model(config=self.cfg, model_path=model_path, device=self.device)
```

Comment:
Also found a bug, no Idea how smoke test was running before for case with loading state_dicts

Reply:
Good point, there is no smoke test checking the model with state_dict, I'll add it in a separate PR

Reply:
It's there actually. But take a look